apollo-router-core/src/tower_compat.rs

```diff
@@ -26,23 +26,24 @@ impl<R> RouterService<R> {
     }
 }
 
-impl<R> tower::Service<RouterRequest> for RouterService<R>
+impl<R> tower::Service<http::Request<Request>> for RouterService<R>
```

Comment:
we should keep the `RouterRequest` type here and have it carry the `Context ` struct. Extensions can apply before the `RouterService` is called (example: JWT validation)

Reply:
That didn't work with `HeaderFilter` because it needs to be mutable and the request is wrapped in an Arc...

I guess we could simply re-create a new RouterRequest (with Request)... :thinking: wdyt?

Reply:
One alternative would be to have a RouterRequest without Context

Reply:
I think RouterRequest was storing the request without an Arc before, so the request could be modified by filters on the frontend, and then it was stored in an Arc before calling the query plan's execute method

Reply:
Before the original request was not part of the Context object I created

Reply:
I am referring to the `RouterRequest` here: https://github.com/apollographql/router/pull/319/commits/57f4bd42893f7dbde0238ecba34a90a04d7b78a6#diff-19c8bc63294ac43b453b868bec15c6dd3facb1f959fc7ddefe7a1fb2683a3c3aL79 that was proposed in https://github.com/apollographql/router/issues/343 and it was the type I used before you introduced the `Context` object.

Reply:
Ok I addressed this here: 75f3b42

(I'm not too fan of the backend/frontend request you'll notice.)

Reply:
that naming was necessary to disambiguate between the request coming from the client (the frontend) and the one going to the subgraph, because both were stored in the `SubgraphRequest` struct. With the request moving into the `Context` struct it should be fine

Reply:
> that naming was necessary
or any other naming that conveys this duality well, we settled with that because we lacked a better (pair of) word(s), I would however be happy to use better names if some come up!

Reply:
No it's fine :sweat_smile: with the current approach I moved it to the Context like Geal said.

Though I do renamed `request` to `http_request` in multiple places to make it clearer that they are `http::Request`.